index.js

```diff
@@ -1,21 +1,23 @@
 'use strict'
 
-var evalAllowed = false;
-try {
-  eval('evalAllowed = true');
-} catch (e) {
-  // eval not allowed due to CSP
-}
+if (!module.hot || process.env.NODE_ENV === 'production') {
+  module.exports = require('./dist/react-hot-loader.production.min.js');
+} else {
+  var evalAllowed = false;
+  try {
+    eval('evalAllowed = true');
+  } catch (e) {
+    // eval not allowed due to CSP
+  }
 
-// RHL needs setPrototypeOf to operate Component inheritance, and eval to patch methods
-var platformSupported = !!Object.setPrototypeOf && evalAllowed;
+  // RHL needs setPrototypeOf to operate Component inheritance, and eval to patch methods
+  var jsFeaturesPresent = typeof window !== 'undefined' && !!Object.setPrototypeOf;
 
-if (!module.hot || process.env.NODE_ENV === 'production' || !platformSupported) {
-  if (module.hot) {
+  if (!jsFeaturesPresent && evalAllowed) {
```

Comment:
Hi! I'm possibly mis-reading this, but should this instead be `(!jsFeaturesPresent || !evalAllowed)` ?

Reply:
🥇 😢 😭 
All this PR is just a disaster.